src/translate/pddl_parser/parsing_functions.py

```diff
@@ -611,15 +610,18 @@ def parse_init(context, alist):
                 if not isinstance(fact, list) or not fact:
                     context.error("Invalid negated fact.", syntax=SYNTAX_LITERAL_NEGATED)
                 atom = pddl.Atom(fact[0], fact[1:])
-                check_atom_consistency(context, atom, initial_false, initial_true, False)
-                initial_false.add(atom)
+                check_atom_consistency(context, atom,
+                                       initial_proposition_values, False)
+                initial_proposition_values[atom] = False
             else:
                 if len(fact) < 1:
```

Comment:
This is a weird test. First, `len(fact) < 1` is a strange way to test for an empty collection. But more importantly, if `fact` can be empty, we will already have crashes in the earlier `if`s that reference `fact[0]`. This is obviously unrelated to the issue, but perhaps it's worth adding a `TODO` here for the future.

Reply:
Looking a bit more, there is already the test `if not fact` earlier that invokes `context.error` so this test already happens earlier and we are safe. So rather than a TODO, I suggest we remove this `if` block as a drive-by cleanup, unless I'm reading the code wrong.

Reply:
This simply makes no sense. Removed.